api/src/main/java/org/apache/iceberg/types/AssignFreshIds.java

```diff
@@ -18,19 +18,13 @@
  */
 package org.apache.iceberg.types;
 
-import java.util.Iterator;
-import java.util.List;
-import java.util.function.Supplier;
 import org.apache.iceberg.Schema;
-import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 
-class AssignFreshIds extends TypeUtil.CustomOrderSchemaVisitor<Type> {
-  private final Schema visitingSchema;
+class AssignFreshIds extends BaseAssignIds {
```

Comment:
The change here is just to refactor to a base class, for logic re-use.

While we can use this class as-is, it seems to blindly re-assign ids, and in our case it seems cleaner to only re-assign ids when they collide.